indy_node/server/action_log.py

```diff
@@ -1,62 +1,196 @@
 import csv
-from abc import ABCMeta
-from datetime import datetime
-from os import path
+import io
+import os
+import datetime
+import functools
+from typing import Type, Union, Tuple
+from enum import Enum, unique
+from dateutil.parser import parse as parse_dt
 
-from dateutil.parser import parse as parse_date
+# TODO
+#  - move csv serializer to separate module
 
 
```

Comment:
yes:

- ` indy_node/test/upgrade/test_upgrade_log.py::test_upgrade_log_loads_legacy_data`
- `indy_node/test/upgrade/test_restart_log.py::test_restart_log_loads_legacy_data`

reason: to operate with objects instead of raw strings after initial parsing (data from extrenal sources, e.g. io)